js/data/buffer.js

```diff
@@ -170,18 +169,16 @@ Buffer.prototype._createPushMethod = function() {
         body += '\nvar ' + offsetId + ' = (offset + ' + attribute.offset + ') / ' + attribute.type.size + ';\n';
 
         for (var j = 0; j < attribute.components; j++) {
-            var valueId = 'value' + i + '_' + j;
+            var rvalue = 'value.' + attribute.name + '[' + j + ']';
             var lvalue = 'this.views.' + attribute.type.name + '[' + offsetId + ' + ' + j + ']';
 
-            body += lvalue + ' = ' + valueId + ';\n';
-
-            argNames.push(valueId);
+            body += lvalue + ' = ' + rvalue + ';\n';
         }
     }
 
     body += '\nreturn index;\n';
 
-    this.push = new Function(argNames, body);
+    this.push = new Function('value', body);
```

Comment:
Yeah, I see the issue. What do you think about changing `push({attrName: [a, b, c]})` to `push('attrName', a, b, c)`, and then calling it multiple times to set multiple attributes? Could this get us to the previous performance while still allowing to selectively add attributes?